syntax = "proto3";

package energy;

// Définition du service
service EnergyService {
  rpc GetEnergyData (EnergyRequest) returns (EnergyResponse);
}

// Ce qu'on envoie (ID du bâtiment)
message EnergyRequest {
  string building_id = 1;
}

// Ce qu'on reçoit (Conso + Statut)
message EnergyResponse {
  string building_id = 1;
  float consumption_kwh = 2;
  string status = 3;  // "Normal", "Surcharge", "Économie"
}